tests/flytekit/unit/types/structured_dataset/test_structured_dataset.py

```diff
@@ -643,3 +660,70 @@ def wf_with_input() -> pd.DataFrame:
 
     pd.testing.assert_frame_equal(wf_no_input(), default_val)
     pd.testing.assert_frame_equal(wf_with_input(), input_val)
+
+
+
+def test_read_sd_from_uri(local_tmp_pqt_file):
+    import os
+
+    os.environ['FLYTE_AWS_ENDPOINT'] = 'http://localhost:30002/'
+    os.environ['FLYTE_AWS_ACCESS_KEY_ID'] = 'minio'
+    os.environ['FLYTE_AWS_SECRET_ACCESS_KEY'] = 'miniostorage'
+
+    @task
+    def upload_pqt_to_s3(local_path: str, remote_path: str) -> None:
+        """Upload local temp parquet file to s3 object storage"""
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            fs = FileAccessProvider(
+                local_sandbox_dir=tmp_dir,
+                raw_output_prefix="s3://my-s3-bucket"
+            )
+            fs.upload(local_path, remote_path)
+
+    @task
+    def read_sd_from_uri(uri: str) -> pd.DataFrame:
+        sd = StructuredDataset(uri=uri, file_format="parquet")
+        df = sd.open(pd.DataFrame).all()
+
+        return df
+
+    @workflow
+    def read_sd_from_local_uri(uri: str) -> pd.DataFrame:
+        df = read_sd_from_uri(uri=uri)
+
+        return df
+
+    @workflow
+    def read_sd_from_remote_uri(
+        local_path: str,
+        remote_path: str,
+        read_uri: str
+    ) -> typing.Tuple[pd.DataFrame, pd.DataFrame]:
+        # Upload parqut to s3
+        upload_pqt_to_s3(local_path=local_path, remote_path=remote_path)
+        ff = FlyteFile(path=REMOTE_PATH)
+        with ff.open(mode="rb") as f:
+            df_s3 = pd.read_parquet(f)
```

Comment:
we shouldn't do this in a workflow, workflow is like a bridge to build your DAG.
task is the place to write Python code.

https://docs.flyte.org/en/latest/user_guide/concepts/main_concepts/workflows.html#divedeep-workflows

<img width="436" alt="image" src="https://github.com/user-attachments/assets/e3b66bdc-057a-454f-a803-15cf78226c30">